src/main/java/br/com/estoque/desafioia/item/Item.java

```diff
@@ -65,4 +67,16 @@ public Integer getQuantidade() {
     public BigDecimal getPreco() {
         return preco;
     }
+
+
+    public void logicaAtualiza(NovoItemRequest request) {
+
+        if (request.nome() != null) this.nome = request.nome();
```

Comment:
aqui vc não ta usando a OO da melhor maneira, o ideal era ter métodos no Request que verificam se os atributos estão presentes ou os getters retornarem Optional, que ai vc faz `request.getNome().irPresent(nome -> this.nome=nome)`.
não é legal operar sobre atributos de outras classes fora delas.